client/src/services/activity.ts

```diff
@@ -41,3 +42,55 @@ export const fetchPCStatus = async () => {
     return error;
   }
 };
+
+export const getRecentActivity = async () => {
+  const url = `${API_URL}/activity/all/recent`;
+  const settings = {
+    method: "GET",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+  };
+
+  try {
+    const response = await fetch(url, settings);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    return error;
+  }
+};
+
+export const useFetchActivities = () => {
+  const store = useBoundStore.getState();
+
+  useEffect(() => {
+    const fetchActivities = async () => {
```

Comment:
This function can be extracted out and made its own function because without polling you'll need to call it individually (not in the `useEffect()`